care/facility/models/patient_consultation.py

```diff
@@ -0,0 +1,133 @@
+from multiselectfield import MultiSelectField
+
+from care.facility.models import CATEGORY_CHOICES, Facility
+from care.facility.models.patient_base import (
+    ADMIT_CHOICES,
+    CURRENT_HEALTH_CHOICES,
+    SYMPTOM_CHOICES,
+    PatientBaseModel,
+    SuggestionChoices,
+)
+from care.users.models import User
+
+
+class PatientConsultation(PatientBaseModel):
+    SUGGESTION_CHOICES = [
+        (SuggestionChoices.HI, "HOME ISOLATION"),
+        (SuggestionChoices.A, "ADMISSION"),
+        (SuggestionChoices.R, "REFERRAL"),
+    ]
+
+    patient = models.ForeignKey("PatientRegistration", on_delete=models.CASCADE, related_name="consultations")
+    facility = models.ForeignKey("Facility", on_delete=models.CASCADE, related_name="consultations")
+    symptoms = MultiSelectField(choices=SYMPTOM_CHOICES, default=1, null=True, blank=True)
+    other_symptoms = models.TextField(default="", blank=True)
+    symptoms_onset_date = models.DateTimeField(null=True, blank=True)
+    category = models.CharField(choices=CATEGORY_CHOICES, max_length=8, default=None, blank=True, null=True)
+    examination_details = models.TextField(null=True, blank=True)
+    existing_medication = models.TextField(null=True, blank=True)
+    prescribed_medication = models.TextField(null=True, blank=True)
+    suggestion = models.CharField(max_length=3, choices=SUGGESTION_CHOICES)
+    referred_to = models.ForeignKey(
+        "Facility", null=True, blank=True, on_delete=models.PROTECT, related_name="referred_patients",
+    )
+    admitted = models.BooleanField(default=False)
+    admitted_to = models.IntegerField(choices=ADMIT_CHOICES, default=None, null=True, blank=True)
+    admission_date = models.DateTimeField(null=True, blank=True)
+    discharge_date = models.DateTimeField(null=True, blank=True)
+
+    def __str__(self):
+        return f"{self.patient.name}<>{self.facility.name}"
+
+    def save(self, *args, **kwargs):
+        if not self.pk or self.referred_to is not None:
+            # pk is None when the consultation is created
+            # referred to is not null when the person is being referred to a new facility
+            self.patient.facility = self.referred_to or self.facility
+            self.patient.save()
+
+        super(PatientConsultation, self).save(*args, **kwargs)
+
+    @staticmethod
+    def has_write_permission(request):
+        if not request.data.get("facility"):
+            # Don't bother, the view shall raise a 400
+            return True
+        return request.user.is_superuser or (
+            request.user.user_type >= User.TYPE_VALUE_MAP["Staff"]
+            and Facility.objects.get(id=request.data["facility"]).created_by == request.user
+        )
+
+    @staticmethod
+    def has_read_permission(request):
+        return True
+
+    def has_object_read_permission(self, request):
+        return request.user.is_superuser or request.user == self.facility.created_by
+
+    def has_object_write_permission(self, request):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5331745)